### PR TITLE
cc regional partner on automated waitlist & decline emails for facili…

### PR DESCRIPTION
**Pull request by tanyaparker**
…tator applications

### CODE REVIEW

**Comment by clareconstantine** [comment]
[The Regional Partner public facing contact is cc’d on 2 emails: when a facilitator application is waitlisted or declined. If there is no public facing contact, cc the first program manager.](https://trello.com/c/4QTi1LTj/793-the-regional-partner-public-facing-contact-is-ccd-on-2-emails-when-a-facilitator-application-is-waitlisted-or-declined-if-there)

File: dashboard/app/mailers/pd/application/facilitator1920_application_mailer.rb [hunk 1]
```diff
@@ -16,21 +16,33 @@ def confirmation(facilitator_application)
     def declined(facilitator_application)
       @application = facilitator_application
 
-      mail(
+      mail_data = {
```
Comment (mehalshah) [inline dashboard/app/mailers/pd/application/facilitator1920_application_mailer.rb]: prefer
```
mail({
  to: ...
  from: ...
  subject: ...
  cc: @application.formatted_partner_contact_email # evaluates to nil if no partner
}.compact)

File: dashboard/app/mailers/pd/application/facilitator1920_application_mailer.rb [hunk 2]
```diff
@@ -16,21 +16,33 @@ def confirmation(facilitator_application)
...
+      end
+
+      mail(mail_data)
     end
 
     def waitlisted(facilitator_application)
       @application = facilitator_application
 
-      mail(
+      mail_data = {
```
Comment (mehalshah) [inline dashboard/app/mailers/pd/application/facilitator1920_application_mailer.rb]: Same here

File: dashboard/app/models/pd/application/application_base.rb [hunk 3]
```diff
@@ -382,6 +382,18 @@ def default_response_score_hash
       {}
     end
 
+    def formatted_partner_contact_email
+      return nil unless regional_partner && regional_partner.contact_email_with_backup.present?
```
Comment (mehalshah) [inline dashboard/app/models/pd/application/application_base.rb]: remove the first conditional, use `regional_partner&.contact_email_with_backup.present?` instead

File: dashboard/test/models/pd/application/facilitator1920_application_test.rb [hunk 4]
```diff
@@ -50,6 +50,41 @@ class Facilitator1920ApplicationTest < ActiveSupport::TestCase
       assert_equal @regional_partner, application.regional_partner
     end
 
+    test 'formatted_partner_contact_email' do
```
Comment (mehalshah) [inline dashboard/test/models/pd/application/facilitator1920_application_test.rb]: These tests would be more appropriate in the application_base test

Comment (mehalshah) [reply dashboard/test/models/pd/application/facilitator1920_application_test.rb]: But not that important

**Review by mehalshah — APPROVED** [review]

**Pull request merged**